src/controls/QskSpinBox.cpp

```diff
@@ -4,356 +4,463 @@
-  connect( this, &S::valueChanged, this, [this](){ polish(); });
+    setBoundaries( 0.0, 1.0 );
+    setAcceptHoverEvents( true );
+    setAcceptedMouseButtons( Qt::LeftButton );
+    setFocusPolicy( Qt::StrongFocus );
+
+    connect( this, &S::focusIndexChanged, this, &S::focusIndicatorRectChanged );
 }
 
 QskSpinBox::~QskSpinBox() = default;
 
-void QskSpinBox::hoverEnterEvent(QHoverEvent* event)
+void QskSpinBox::hoverEnterEvent( QHoverEvent* const event )
 {
-  m_data->saveMousePosition( qskHoverPosition( event ) );
+    m_data->saveMousePosition( qskHoverPosition( event ) );
 }
 
-void QskSpinBox::hoverLeaveEvent(QHoverEvent* event)
+void QskSpinBox::hoverLeaveEvent( QHoverEvent* /*const event */ )
 {
-  m_data->saveMousePosition( {} );
+    m_data->saveMousePosition( {} );
 }
 
-void QskSpinBox::hoverMoveEvent(QHoverEvent *event)
+void QskSpinBox::hoverMoveEvent( QHoverEvent* const event )
 {
-  m_data->saveMousePosition( qskHoverPosition( event ) );
+    m_data->saveMousePosition( qskHoverPosition( event ) );
 }
 
-void QskSpinBox::mouseReleaseEvent(QMouseEvent *event)
+void QskSpinBox::mouseReleaseEvent( QMouseEvent* const event )
 {
-  m_data->saveMousePosition( qskMousePosition( event ) );
+    m_data->saveMousePosition( qskMousePosition( event ) );
 
-  const auto focus = ( focusPolicy() & Qt::ClickFocus ) == Qt::ClickFocus && !QGuiApplication::styleHints()->setFocusOnTouchRelease();
+    const auto focus = m_data->focusNow();
 
-  if(subControlRect(QskSpinBox::Inc).contains( event->pos() ))
-  {
-    increment(+stepSize());
-
-    if( focus )
+    if ( subControlRect( QskSpinBox::IncrementPanel ).contains( event->pos() ) )
     {
-      m_data->setFocus(PrivateData::Inc);
-    }
+        increment( +stepSize() );
 
-    return;
-  }
+        if ( focus )
+        {
+            m_data->setFocus( Increment );
+        }
 
-  if(subControlRect(QskSpinBox::Dec).contains( event->pos() ))
-  {
-    increment(-stepSize());
+        return;
+    }
 
-    if( focus )
+    if ( subControlRect( QskSpinBox::DecrementPanel ).contains( event->pos() ) )
     {
-      m_data->setFocus(PrivateData::Dec);
-    }
+        increment( -stepSize() );
 
-    return;
-  }
+        if ( focus )
+        {
+            m_data->setFocus( Decrement );
+        }
 
-  if(subControlRect(QskSpinBox::TextPanel).contains( event->pos() ))
-  {
-    if( focus )
-    {
-      m_data->setFocus(PrivateData::Text);
+        return;
     }
 
-    return;
-  }
+    if ( subControlRect( QskSpinBox::TextPanel ).contains( event->pos() ) )
+    {
+        if ( focus )
+        {
+            m_data->setFocus( Textbox );
+        }
 
-  event->ignore();
+        return;
+    }
+
+    event->ignore();
 }
 
-void QskSpinBox::mousePressEvent(QMouseEvent *event)
+void QskSpinBox::mousePressEvent( QMouseEvent* const event )
 {
-  m_data->saveMousePosition( -1 * qskMousePosition( event ) );
+    m_data->saveMousePosition( -1 * qskMousePosition( event ) );
```

Comment:
How can we do this more elegantly? Currently -pos => pressed +pos == hovered and pos{} == leave

Maybe somethin like:

```cpp
q->setSkinHint( QskSpinBox::Layout | QskAspect::Flag | QskAspect::Position, Qt::MouseButtonPress );
```

We could also set the subcontrol's state in the QskSpinBox, but this feels wrong to have two locations manipulating states. Currently all/the majority of states are being manipulated by the skinlet.